README.md

```diff
@@ -16,86 +16,77 @@ The **Hyphen Toggle OpenFeature Web Provider** is an OpenFeature provider implem
-        role: 'admin',
-      },
-    },
-  };
-
-  useEffect(() => {
-    const setupOpenFeature = async () => {
-      try {
-        await OpenFeature.setProviderAndWait(new HyphenProvider(publicKey, options));
-        await OpenFeature.setContext(context);
-
-        const client = OpenFeature.getClient();
-        const data = client.getNumberDetails('your-flag-key', 0);
-        setFeatureFlagValue(data.value);
-      
-      } catch (error) {
-        console.error('Error setting up OpenFeature:', error);
-      }
-    };
-
-    setupOpenFeature();
-  }, []);
-
-
-  return (
-    <>
-      <h1>OpenFeature Example</h1>
-      <p>Feature flag value: {featureFlagValue}</p>
-    </>
-  );
-}
-
-export default App
-```
-
+1. **Configure the Provider**: Register the `HyphenProvider` with OpenFeature using your `publicKey` and provider options.
+
+    Add the following setup to your application:
+   ```typescript jsx
+   import { OpenFeature, ProviderEvents } from '@openfeature/web-sdk';
+   import { HyphenProvider, HyphenProviderOptions } from '@hyphen/openfeature-web-provider';
+   
+   const publicKey = 'your-public-key'; // Replace with your Hyphen publicKey
+   
+   const options: HyphenProviderOptions = {
+     application: 'your-app-name', // Replace with your application name
+     environment: 'production',    // Replace with the appropriate environment
+   };
+   
+   OpenFeature.setProvider(new HyphenProvider(publicKey, options));
+   
+   createRoot(document.getElementById('root')!).render(<App />);
+   ```
+
+2. **Set Up the context**: Use ``OpenFeature.setContext`` to configure the context needed for feature targeting. This context should include relevant user information, typically obtained from an authentication process.
+   ```typescript jsx
+   const AuthContext = createContext({ isLoading: true, user: null });
+
+   export const MockAuthProvider = ({ children }: { children: React.ReactNode }) => {
+      const [authState, setAuthState] = useState({ isLoading: true, user: null });
+   
+      useEffect(() => {
+        setTimeout(() => {
+          const user = {
+            id: 'user-123',
+            name: 'John Doe',
+            email: 'user@example.com',
+            role: 'admin',
+          };
+          OpenFeature.setContext({
+            targetingKey: user.id,
+            customAttributes: { role: user.role }, // Additional targeting attributes
+          });
+   
+         setAuthState({ isLoading: false, user });
+          }, 1000);
+      }, []);
+   
+      return <AuthContext.Provider value={authState}>{children}</AuthContext.Provider>;
+      };
+   
+      export const useAuth = () => useContext(AuthContext);
+   ```
+   
+   
+3. **Evaluate Feature Flags**: Use the `OpenFeature` client to evaluate feature flags in your application.
+
+    ```typescript jsx
+    const client = OpenFeature.getClient();
+    const isEnabled = client.getStringDetails('your-flag-key', 'default value');
```

Comment:
Let's just show the setContext call. They can go into the example to see the whole thing.